src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/SettingsTests.tsx

```diff
@@ -32,64 +33,74 @@ import {EmptyTestsContainer, StyledOptionWrapper, StyledStepsList} from './Setti
 
 const {Option} = Select;
 
-const SettingsTests = () => {
+const SettingsTests: React.FC = () => {
   const {isClusterAvailable} = useContext(MainContext);
   const {entityDetails} = useContext(EntityDetailsContext);
+  const typedEntityDetails = entityDetails as TestSuite;
+
   const mayEdit = usePermission(Permissions.editEntity);
 
   const [isDelayModalVisible, setIsDelayModalVisible] = useState(false);
 
-  const scrollRef = useRef(null);
+  const scrollRef = useRef<HTMLDivElement>(null);
 
   const executors = useAppSelector(selectExecutors);
 
   const {data: testsList = []} = useGetTestsListForTestSuiteQuery(entityDetails.name, {skip: !isClusterAvailable});
   const {data: allTestsList = []} = useGetAllTestsQuery(null, {skip: !isClusterAvailable});
   const [updateTestSuite] = useUpdateTestSuiteMutation();
 
-  const testsData = useMemo<any[]>(() => {
-    return testsList.map((item: Test) => ({
+  const testsData: TestForTestSuiteSteps[] = useMemo(() => {
+    return testsList.map(item => ({
       name: item.name,
       namespace: item.namespace,
       type: getTestExecutorIcon(executors, item.type),
     }));
   }, [testsList]);
 
-  const allTestsData = useMemo(() => {
-    return allTestsList.map((item: TestWithExecution) => ({
+  const allTestsData: TestForTestSuiteSteps[] = useMemo(() => {
+    return allTestsList.map(item => ({
       name: item.test.name,
       namespace: item.test.namespace,
       type: getTestExecutorIcon(executors, item.test.type),
     }));
   }, [allTestsList]);
 
-  const initialSteps = useMemo(
+  const initialSteps: (TestSuiteStepWithDelay | TestSuiteStepWithExecute)[] = useMemo(
     () =>
-      (entityDetails?.steps || []).map((step: any) => {
-        if (step.delay) {
-          return {
-            ...step,
-            id: nanoid(),
-          };
-        }
-        return {
-          ...step,
-          id: nanoid(),
-          execute: {
-            ...step.execute,
-            type: testsData.find(item => item.name === step.execute.name)?.type,
-          },
-        };
-      }),
-    [entityDetails?.steps, testsData]
+      typedEntityDetails.steps
```

Comment:
Why not the previous solution - `entityDetails?.steps || []`?

Reply:
Needed type for `entityDetails`, your approach works good here